src/compact_filter.cc

```diff
@@ -108,7 +110,12 @@ bool SubKeyFilter::Filter(int level,
                                   std::string *new_value,
                                   bool *modified) const {
   InternalKey ikey(key, stor_->IsSlotIdEncoded());
-  bool result = IsKeyExpired(ikey, value);
+  Metadata metadata(kRedisNone, false);
+  if (GetMetadata(ikey, &metadata)) {
+    return true;
+  }
```

Comment:
I don't think the name `GetMetadata` is clear, so I'm confused here.

Reply:
I didn't invent a better name. `FindMetadata`, `FillMetadata`?

Reply:
Let me think about it. I still don't think the logic here is clear enough.

Reply:
this function is looking for metadata and fills it, also returned true if kv can be deleted.

Reply:
So it does more than one thing, and we should separate those things.

Reply:
Hm, I don't know how I can split it. 
there are 4 actions:
- get_db
- build_metadata_key
- find_metdata
- decode_metadata

 And all of the actions need to get metadata..

Reply:
I thought about it a little more, `GetMetadata` is OK.

Reply:
@tufitko @caipengbo I think another idea is `GetMetadata` returns Status instead of bool, so that caller determine whether to keep the subkey/blob or not. Not sure if I clarify clearly enough.

Reply:
The pseudo-code should be like below:

```
Filter() {

  Status s = GetMetadata(ikey, &metadata);
  if (s.IsNotFound()) {
      return true;
  } 
  if (!s.IsOK()) {
     // log error
     return false;
  } 
  ... check is expired or other logic here
}
```

Reply:
> @tufitko @caipengbo I think another idea is GetMetadata returns Status instead of bool, so that caller determine whether to keep the subkey/blob or not. Not sure if I clarify clearly enough.

Yeah, that makes it clearer!

Reply:
Oh, okey, it’s good, I will be commited later

Reply:
❤️ Thanks